### ISSUE TITLE
download speed drops to 1KB/s after a while

### ISSUE THREAD

**Issue by ghost** [kzahel/jstorrent-legacy-app#59]
Hi !

I just bought your app yesterday, however i am not able to fully download a file.
In fact the first ~30% works perfectly fine but after that the download speed drops down to 1KB/s, and it is the same with every torrent and every download location.

![screenshot 2014-08-02 at 18 37 46](https://cloud.githubusercontent.com/assets/6660124/3788409/ca95064e-1a64-11e4-9121-8e0075067661.png)

Even if i pause, stop and start again the download speed won't go up again.

Regards.

**Comment by kzahel** [comment]
Hm. You have 200 peers? Try with a lower setting first maybe... like around 50? Especially if you have a lower end chromebook.

**Comment by parrilla2989** [comment]
any luck with download speed?

**Comment by ghost** [comment]
Yeah well i tried 200 to see whether it changes something or not but with any number it is still the same.
It drops down after 20% or 50%, sometimes 80% to 1KB/S and never goes up...

My chromebook is a toshiba chromebook.

Regards,

**Comment by knicolle** [comment]
I bought this for my asus chromebox a few days ago, and [maybe this is known or I am missing something] the download speed is fast when the app is minimised, but crawls to 1kb/s when visible.

**Comment by kzahel** [comment]
@knicolle that's very good information. I am still trying to reproduce the slowdown. Does this only happen after downloading a lot of data (like > 2GB?)

**Comment by knicolle** [comment]
Hi - in my experience, this has not been size dependent; as soon as a torrent is loaded and connects to peers and starts downloading the speed increases and then degrades very quickly.  If the app is minimised the speed picks up immediately, and then if the app is restored to view [the speed can bee seen as very high for a few seconds], then the speed degrades to 1kb/s ish.

I have also noticed that even when downloading in a minimised state, the connection tends to stop after 2GB ish, and has to be stopped and restarted; but I have no idea if this is related to the overall speed issue of minimised versus maximised.

**Comment by kzahel** [comment]
@knicolle Perhaps your computer is being slowed down by the constant updates inside the "peers" tab. When the view is minimized, it is not constantly re-drawing the bytes downloaded from each peer, etc. Can you verify that if you switch out of the "peers" tab into say the "info" tab (which does not update and cause redraw) that the download continues at normal speed?

If this is the case, I will have to throttle the re-drawing to something like only once per second, instead of each time the data changes, at least for low end devices

**Comment by knicolle** [comment]
Ok....let me check that....but I don't think it is the case as quite often I have my view on the files tab.  But I will check what happens if I am on the 'info' tab.

FYG - I have the Asus Chromebox i3 1.7 ghz with 4gb ram, so it is not really a low end machine; in fact is one of the higher end Chrome machines?

**Comment by knicolle** [comment]
Just a though...perhaps on the peers tab, rather than continuously update the bytes downloaded, just provide each peer's speed they are providing?

As a side question - what determines if you are choked?  As I seem to get choked/blocked a lot - I don't have the 'cloak' ticked.

**Comment by kzahel** [comment]
Yes, that is a good idea.

Being unchoked is determined by the other peer. You'll be unchoked when
your peer has enough upload slots available basically.

**Comment by knicolle** [comment]
Hi.....checked....and it doesn't matter what tab I look at, the speed degrades very quickly when the app is visible....I was just on the info tab...minimised and then restored - speed went from 1MB/s to 1KB/s in about 5 seconds.

**Comment by kzahel** [comment]
Yup I've finally confirmed this bug. I'm hoping it isn't a chrome platform bug. I'm going to see if I can write some tests that reproduce it.

**Comment by knicolle** [comment]
Understood…..by the way, I forgot to say I use the beta version of Chrome; is that relevant?

**Comment by knicolle** [comment]
Have you manage to fix the bug?  Or work out if it is a Chrome Platform Bug?

I still have to minimise your app to get the spped high - otherwise it drops again to 1kb/s

**Comment by Mephistofleas** [comment]
Hello Acer C720 user with same issue. Degrades from 1/3/5 MB to as low as 1kb/s and sometimes lower when viewing the app. Obviously not the end of the world as it 'goes away' by minimising. 

Do you believe it to be hardware related? Could you perhaps explain it a bit simpler :-)

**Comment by kzahel** [comment]
I'm still looking into the cause. I think it's a memory leak related to the
UI framework I used. That would explain why minimizing alleviates the
problem. I have seen though that it is still slow (<1Kb) even after
minimizing. I have to reboot the app entirely. I'll see if I can reproduce
your case. That would be very helpful! Thanks for the feedback

**Comment by knicolle** [comment]
Hi

I still get the <1k error, and minimising does still help – but over time it eventually stops and I have to stop and start the Torrent.

I have to reboot the app every few days, but I am always running the app with live Torrents.

**Comment by kzahel** [comment]
I identified a lot of problems that were causing the downloads to stop. The latest release should fix them all.

**Comment by ghost** [comment]
Thanks !
I'll install it and try it again sometimes, i will keep you in touch.

2016-04-14 4:25 GMT+02:00 Kyle Graehl notifications@github.com:

**Comment by jpcorreab** [comment]
I am having the same problem torrents download good and fast for a while and then speed drops as low as a few B/s. In my case it does not go away by minimizing the app (it does get better but still really slow), but changing to another wifi network.
It is quite odd because if I change to another networks it works fine, but if I change again to the one I was using before, it drops inmediatly. I have already checked and it is not a network problem, as other apps like netflix worked with good speed. It is not a router or firewall problem either as the same network works fine downloading torrents with my windows computer.
What even more strange is that the first network is the fast one and the other, as it is my cellphone sharing wifi, has a really slower speed.

**Comment by saiki-k** [comment]
I'm facing the exact same issue as @jpcorreab Any leads on the cause? @kzahel

**Comment by knicolle** [comment]
Hi - still only happens for me when the app is maximised or restored; when minimised the download speeds are blisteringly fast.

I check check occasionally by maximising the app to see how they are going [the speed drops accordingly] then minimise again.

**Comment by kzahel** [comment]
The minimized is faster is true, if you have the Peers tab or some other very "busy" tab (with many values changing constantly) open. But if you are in the info tab, the actual download speed won't be any different. It's only immediately after you un-minimize that the value looks large. This is just due to my only calculating average download speed based on the last 2 seconds or so, and since it hasn't calculated this since it was last minimized, the value "consumes" the previously downloaded data over the last several minutes and it looks large.

**Comment by kzahel** [comment]
The legacy Chrome App is no longer maintained. I've rebuilt JSTorrent from scratch — the new version can hit 90 MB/s on mobile and 100+ MB/s on desktop. Get it at **[jstorrent.com](https://jstorrent.com)**

If you run into any issues, please leave feedback!
